### PR TITLE
Sprin 1936/gtm events for noah language learning

### PR DESCRIPTION
**Pull request by gauri-smaowl**

### CODE REVIEW

File: src/app/(dashboard)/(routes)/languages/components/topic-card.tsx [hunk 1]
```diff
@@ -199,16 +240,17 @@ export default function TopicCard({
...
-                    `/languages/quiz?lang=${lang}&topic=${
-                      topic.id
-                    }&level=${levelId}&cards=${
-                      hasCompletedAllStates
-                        ? selectedState || getStateWithLowestPoints()
-                        : nextState
-                    }`
-                  )
+                onClick={handleClickForQuiz
+                  // () =>
```
Comment (gauri-smaowl) [inline src/app/(dashboard)/(routes)/languages/components/topic-card.tsx]: Remove commented code

File: src/app/(dashboard)/(routes)/subject-dashboard/page.tsx [hunk 2]
```diff
@@ -76,7 +77,7 @@ const PageContent = () => {
       eventAction: "subject_opened",
       label: userId ? "student" : "guest",
       label1: userId || null,
-      label2: quizPath,
+      label2: subject=="mathematics" ? "math" :subject?.toLocaleLowerCase()||"",
```
Comment (gauri-smaowl) [inline src/app/(dashboard)/(routes)/subject-dashboard/page.tsx:80]: Why is this changed?

File: src/app/(dashboard)/(routes)/languages/components/topic-card.tsx [hunk 3]
```diff
@@ -180,16 +220,17 @@ export default function TopicCard({
...
-                    `/languages/learn?lang=${lang}&topic=${
-                      topic.id
-                    }&level=${levelId}&cards=${
-                      hasCompletedAllStates
-                        ? selectedState || getStateWithLowestPoints()
-                        : nextState
-                    }`
-                  )
+                onClick={handleLearnButtonClick
+                  // () =>
```
Comment (gauri-smaowl) [inline src/app/(dashboard)/(routes)/languages/components/topic-card.tsx]: Remove commented code

File: src/app/(dashboard)/(routes)/languages/components/topic-card.tsx [hunk 4]
```diff
@@ -94,6 +96,44 @@ export default function TopicCard({
...
+  const handleLearnButtonClick = () => {
+    saveGTMEvents({
+      eventAction: "learn_language_opened",
+      label: userId?"Student":"Guest",
+      label1: userId?.toString()||null,
+      label2: lang,
+      label3: topic.name||null,
+      label4: null,
+    });
+    router.push(
```
Comment (gauri-smaowl) [inline src/app/(dashboard)/(routes)/languages/components/topic-card.tsx:110]: put consolelog on each of these fields so that QA can verify it on staging

File: src/app/(dashboard)/(routes)/subject-dashboard/page.tsx (hunk 2 above)
Comment (dipendra-28) [reply src/app/(dashboard)/(routes)/subject-dashboard/page.tsx:80]: this quizpath have values of all the subjects not the newly added  languages , that's why I changed this
